_posts/2022-07-01-argocd-image-updater-for-functions.md

```diff
@@ -0,0 +1,248 @@
+  namespace: argocd
+spec:
+  destination:
+    namespace: openfaas
+    server: {{ .Values.spec.destination.server }}
+  project: default
+  source:
+    helm:
+      parameters:
+      - name: operator.create
+        value: "true"
+      - name: generateBasicAuth
+        value: "true"
+      - name: functionNamespace
+        value: openfaas-fn
+    path: chart/openfaas
+    repoURL: https://github.com/openfaas/faas-netes.git
+  syncPolicy:
+    automated:
+      selfHeal: true
+```
+
+The template for the openfaas-functions application:
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: openfaas-functions
+  namespace: argocd
+spec:
+  destination:
+    server: {{ .Values.spec.destination.server }}
+  project: default
+  source:
+    path: chart/functions
+    repoURL: https://github.com/welteki/openfaas-argocd-example.git
+  syncPolicy:
+    automated:
+      selfHeal: true
+```
+
+The full example is [available on GitHub](https://github.com/welteki/openfaas-argocd-example)
+
+Once 
+The parent app can be created and synced via the CLI:
+```bash
+argocd app create applications \
+    --dest-namespace argocd \
+    --dest-server https://kubernetes.default.svc \
+    --repo https://github.com/welteki/openfaas-argocd-example.git \
+    --path chart/applications
+
+argocd app sync applications
+```
+
+You can check the Argo CD UI to see if the apps were synced successfully or use `kubectl` to verify if OpenFaaS and the functions are running.
+
+```bash
+kubectl get pods -n openfaas
+
+NAME                              READY   STATUS    RESTARTS          AGE
+alertmanager-c4df79ff7-2fxcs      1/1     Running   2 (4h42m ago)     39h
+basic-auth-plugin-588f588-rkgmz   1/1     Running   2 (4h42m ago)     39h
+gateway-55fd54cb76-xf4n7          2/2     Running   3 (4h42m ago)     39h
+nats-67d8f684f8-x46zj             1/1     Running   2 (4h42m ago)     39h
+prometheus-cd4844fc7-c6j2b        1/1     Running   2 (4h42m ago)     39h
+queue-worker-5795ff9bb5-tkwpv     1/1     Running   3 (4h42m ago)     39h
+```
+
+```bash
+kubectl get functions -n openfaas-fn
+
+NAME             IMAGE
+email-notify     welteki/email-notify:0.1.0
+marketing-list   welteki/marketing-list:0.1.0
+```
+
+### Install and configure the ArgoCD Image Updater
+We install the updater as Kubernetes workload in the Argo CD namespace.
+
+```bash
+kubectl apply -n argocd  \
+    -f https://raw.githubusercontent.com/argoproj-labs/argocd-image-updater/stable/manifests/install.yaml
+```
+
+The openfaas-functions application needs to be correctly annotated for the Argo CD image Updater to know which application and images it should inspect and update.
+
+The following annotation were added to the openfaas-functions application template:
+```yaml
+annotations:
+    argocd-image-updater.argoproj.io/image-list: marketingList=welteki/marketing-list:~0.1
+    argocd-image-updater.argoproj.io/marketingList.helm.image-spec: marketingList.image
+```
+
```

Comment:
Explain the two annotations in more detail

Explain what kinds of things you can put in `image-list`, and what `~0.1` means.

Reply:
I updated this section